src/Microsoft.ML.Ensemble/EnsembleUtils.cs

```diff
@@ -31,7 +31,7 @@ public static RoleMappedData SelectFeatures(IHost host, RoleMappedData data, Bit
             var name = data.Schema.Feature.Name;
             var view = LambdaColumnMapper.Create(
                 host, "FeatureSelector", data.Data, name, name, type, type,
-                (ref VBuffer<Single> src, ref VBuffer<Single> dst) => SelectFeatures(ref src, features, card, ref dst));
+                (ref VBuffer<Single> src, ref VBuffer<Single> dst) => SelectFeatures(in src, features, card, ref dst));
```

Comment:
>ref [](start = 17, length = 3)

is it worth to modify ValueMapper to switch from `public delegate void ValueMapper<TSrc, TDst>(ref TSrc src, ref TDst dst);` to 
`public delegate void ValueMapper<TSrc, TDst>(in TSrc src, ref TDst dst);` ?

Reply:
Hmmm, yes, this seems like a good idea. Indeed in a change yesterday I see @eerhardt already did something like this by improving `RefPredicate` to `InPredicate` in #1405, and this seems quite similar.

Reply:
I can do this in a follow up PR. This change was necessary first, since there are a ton of places where `ValueMapper` implementations call into these methods. Once these are in, `ValueMapper` can be updated as well.